### PR TITLE
Remove tracking codes from shareable URLs in IDE and browser extensions

### LINKED ISSUE #41987: VS Code: Shared links are inconveniently long
Problem description by @olafurpg: "I used the "Copy Sourcegraph File Link" command in VS Code to [share a link in an issue]( https://github.com/sourcegraph/sourcegraph/issues/41979#issuecomment-1255950216) and I was put off by the noise in the URL itself. In this case, I feel we've prioritized collecting metrics at the cost of providing a good user experience. I don't want to use this command because the URLs have too much unrelated information

Solution suggestion by @philipp-spiess: "What Twitter does is pretty nice they have a short-form tracking code that the server can translate to the longer form UTM campaigns, e.g.: "https://twitter.com/fuckedupfoods/status/1573103725673324544?s=20&t=rZcKxySHqomMZyBuW6eFBQ" – the `s=20` part is enough here to add the tracking they needed

image

### LINKED ISSUE #41988: JetBrains: Shared links are inconveniently long
This bug report was originally for VS Code (here: https://github.com/sourcegraph/sourcegraph/issues/41987), but it also applies to JetBrains. Some part of the fix (unpacking short URLs on the back end) will be shared between IDE extensions, but that back end feature must also be used in both extensions separately, so I'm also adding this issue.

For any discussion and problem description, see the VS Code issue: https://github.com/sourcegraph/sourcegraph/issues/41987

Quick summary: When using the "Copy Sourcegraph Link", the URL is quite long, and it provides a bad UX. Twitter solves it pretty nicely with a short-form tracking code that the server can translate to the longer form UTM campaigns.

### PR DESCRIPTION
**Pull request by philipp-spiess**
Closes #41987
Closes #41988

We're removing all UTM campaign parameters from the IDE and browser extensions. Time for cleaner URLs! Some context on the decision [in this Slack thread](https://sourcegraph.slack.com/archives/C01LZKLRF0C/p1663923103346749).

I've conveniently combined all the changes into one PR but I can split it up if someone wants 😅.

### Browser extension

https://www.loom.com/share/988eb32903474ded9ac42a4eec266831

### VS Code

https://www.loom.com/share/e2021b0d5668447ca1c7fe3949c66c8e

### JetBrains

https://www.loom.com/share/1a6b4a96270747d0aa09ffc7835e90d1

## App preview:

- [Web](https://sg-web-ps-remove-shareable-urls-tracking.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-dowqloaveg.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Review by taras-yemets — APPROVED** [review]
Thanks for taking care of it, Philipp!

File: client/jetbrains/src/main/java/com/sourcegraph/website/URLBuilder.java [hunk 1]
```diff
@@ -48,18 +47,12 @@ public static String buildSourcegraphBlobUrl(@NotNull Project project, @NotNull
...
-    private static String buildMarketingParams() {
-        String productName = ApplicationInfo.getInstance().getVersionName();
-        String productVersion = ApplicationInfo.getInstance().getFullVersion();
-
+    private static String buildVersionParams() {
         return "editor=" + URLEncoder.encode("JetBrains", StandardCharsets.UTF_8)
-            + "&version=v" + URLEncoder.encode(ConfigUtil.getPluginVersion(), StandardCharsets.UTF_8)
-            + "&utm_product_name=" + URLEncoder.encode(productName, StandardCharsets.UTF_8)
-            + "&utm_product_version=" + URLEncoder.encode(productVersion, StandardCharsets.UTF_8);
+            + "&version=v" + URLEncoder.encode(ConfigUtil.getPluginVersion(), StandardCharsets.UTF_8);
```
Comment (vdavid) [inline client/jetbrains/src/main/java/com/sourcegraph/website/URLBuilder.java:56]: I don't think we use the "editor" and "version" params either? How about we go all the way and:
a.) remove these too
OR
b.) use something very short like like "e=JetBrains2.0.0" ("e" for editor)

**Review by vdavid — APPROVED** [review]
The changes look good to me. Added +1 suggestion.

File: client/jetbrains/src/main/java/com/sourcegraph/website/URLBuilder.java (hunk 1 above)
Comment (philipp-spiess) [reply client/jetbrains/src/main/java/com/sourcegraph/website/URLBuilder.java:56]: Unfortunately those two params are required by our backend (I documented it in the video if you're curious). Even if we update the backend we'd still have to support old instances so this was not ergonomic to do right away 😒

Comment (vdavid) [reply client/jetbrains/src/main/java/com/sourcegraph/website/URLBuilder.java:56]: Gotcha. Thanks, and sorry, I missed the video 😬

Comment (philipp-spiess) [reply client/jetbrains/src/main/java/com/sourcegraph/website/URLBuilder.java:56]: No worries, I should have written about it but I was in a hurry 😫

**Review by abeatrix — APPROVED** [review]
Much cleaner 😎 thanks for the quick turnaround!

**Comment by taras-yemets** [comment]
@philipp-spiess, let's include these changes in the VSCE [changelog](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/vscode/CHANGELOG.md).

**Pull request merged**